### PR TITLE
update TestRunWithKernelMemory

### PR DESCRIPTION
**Pull request by sunyuan3**
Signed-off-by: Yuan Sun sunyuan3@huawei.com

### CODE REVIEW

File: integration-cli/docker_cli_run_unix_test.go [hunk 1]
```diff
@@ -205,13 +205,37 @@ func (s *DockerSuite) TestRunWithCpuPeriod(c *check.C) {
...
+	if out != "102400000" {
+		c.Fatalf("setting the kernel memory limit failed")
+	}
+
+	dockerCmd(c, "run", "--kernel-memory", "1G", "--name", "test3", "busybox", "true")
+
+	out, err = inspectField("test3", "HostConfig.KernelMemory")
+	c.Assert(err, check.IsNil)
+	if out != "1073741824" {
+		c.Fatalf("setting the kernel memory limit failed")
```
Comment (runcom) [inline integration-cli/docker_cli_run_unix_test.go]: could you provide expected and actual values here so it'll be easier to see which failed?

File: integration-cli/docker_cli_run_unix_test.go [hunk 2]
```diff
@@ -205,12 +205,36 @@ func (s *DockerSuite) TestRunWithCpuPeriod(c *check.C) {
...
+
+	dockerCmd(c, "run", "--kernel-memory", "100000K", "--name", "test2", "busybox", "true")
+
+	out, err = inspectField("test2", "HostConfig.KernelMemory")
+	c.Assert(err, check.IsNil)
+	if out != "102400000" {
+		c.Fatalf("setting the kernel memory limit failed, expected 102400000, actual %q", out)
+	}
+
+	dockerCmd(c, "run", "--kernel-memory", "1G", "--name", "test3", "busybox", "true")
```
Comment (runcom) [inline integration-cli/docker_cli_run_unix_test.go]: Are there any reasons why this test (1G) and the one above (100000K) are needed?
The first one (50M) is covering positive values and the one below negative with error.
Having the correct value converted between sizes (k,m,g to bytes) is covered with unit tests in pkg/units so I don't think these two tests are needed

**Comment by sunyuan3** [comment]
HI @runcom , the code has been updated according to your suggestion.

**Comment by runcom** [comment]
LGTM if janky goes well

File: integration-cli/docker_cli_run_unix_test.go [hunk 3]
```diff
@@ -205,12 +205,20 @@ func (s *DockerSuite) TestRunWithCpuPeriod(c *check.C) {
 func (s *DockerSuite) TestRunWithKernelMemory(c *check.C) {
 	testRequires(c, kernelMemorySupport)
 
-	dockerCmd(c, "run", "--kernel-memory", "50M", "--name", "test", "busybox", "true")
+	dockerCmd(c, "run", "--kernel-memory", "50M", "--name", "test1", "busybox", "true")
 
-	out, err := inspectField("test", "HostConfig.KernelMemory")
+	out, err := inspectField("test1", "HostConfig.KernelMemory")
 	c.Assert(err, check.IsNil)
 	if out != "52428800" {
```
Comment (cpuguy83) [inline integration-cli/docker_cli_run_unix_test.go]: Note that this should really be `c.Assert(out, check.Equals, "52428800")`, it makes the error output nicer and doesn't need the extra comment.

File: integration-cli/docker_cli_run_unix_test.go [hunk 4]
```diff
@@ -205,12 +205,20 @@ func (s *DockerSuite) TestRunWithCpuPeriod(c *check.C) {
...
-		c.Fatalf("setting the kernel memory limit failed")
+		c.Fatalf("setting the kernel memory limit failed, expected 52428800, actual %q", out)
+	}
+
+	out, _, err = dockerCmdWithError("run", "--kernel-memory", "-16m", "--name", "test2", "busybox", "echo", "test")
+	expected := "invalid size"
+	c.Assert(err, check.NotNil)
+
+	if !strings.Contains(out, expected) {
```
Comment (cpuguy83) [inline integration-cli/docker_cli_run_unix_test.go]: `c.Assert(out, checker.Contains, expected)`

**Comment by jfrazelle** [comment]
LGTM

**Comment by jfrazelle** [comment]
```

**Comment by sunyuan3** [comment]
Hi @jfrazelle , many thanks for your reminding.

**Comment by sunyuan3** [comment]
Hi @jfrazelle , Could you please trigger the test again?

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**